lib/Doctrine/ORM/Query/ParameterTypeInferer.php

```diff
@@ -53,7 +53,11 @@ public static function inferType($value)
             return Type::BOOLEAN;
         }
 
-        if ($value instanceof \DateTime || $value instanceof \DateTimeInterface) {
+        if ($value instanceof \DateTimeImmutable) {
```

Comment:
Why is this failing? The changed code used fully qualified names as well.

Reply:
Try to use `use` instead of having the FQCN directly in the code. The same with your other changes.

Reply:
Well, yeah, that's obvious. My question is _why_? Because it doesn't make any sense.

The original code uses FQN and evidently don't adhere to the set coding standards. Why force the CS for the modified code then?

Now, I have two options - either I'll add `use`s only for the changed lines leaving the modified files inconsistent (one line will have, for instance, `DateTimeInterface` while another `\DateInterval`), or I will add `use`s for all FQN in all modified files making the surface of the PR much bigger with a lot of changes not related to the proposed change. And both options are wrong.

Reply:
ORM is a long running project and some parts, like the coding styles were introduced/updated late over time. This is why there is a coding style check that targets changed code that are part of a commit/diff. In your case the changes in the lines with `instanceof` are now affected by the style-checks and need to be adapted in this PR.

Reply:
That's all nice but I still fail to see any benefits of that as it only clutters the code, must make merging between branches harder and leads to useless discussions like this one. IMO all active branches should be updated to follow CS rules and then makes sense to force CS checks in PRs. But whatever works for Doctrine maintainers, take this just as feedback from a random contributor that current state makes contribution unnecessary harder…

The code has been updated, all checks are green now.